autode/opt/optimisers/base.py

```diff
@@ -0,0 +1,608 @@
+            raise ValueError('Tolerance on the gradient (||∇E||) must be '
+                             f'positive. Had: gtol={value}')
+
+        self._gtol = GradientRMS(value)
+
+    @property
+    def etol(self) -> PotentialEnergy:
+        """
+        Energy tolerance between two consecutive steps of the optimisation
+
+        -----------------------------------------------------------------------
+        Returns:
+            (autode.values.PotentialEnergy): Energy tolerance
+        """
+        return self._etol
+
+    @etol.setter
+    def etol(self, value: Union[int, float, PotentialEnergy]):
+        """Set the energy tolerance"""
+        if float(value) <= 0:
+            raise ValueError('Tolerance on the energy change is absolute so '
+                             f'must be positive. Had etol = {value}')
+
+        self._etol = PotentialEnergy(value)
+
+    @classmethod
+    def optimise(cls,
+                 species: 'autode.species.Species',
+                 method:  'autode.wrappers.base.Method',
+                 maxiter: int = 100,
+                 gtol:    Union[float, GradientRMS] = GradientRMS(1E-3, units='Ha Å-1'),
+                 etol:    Union[float, PotentialEnergy] = PotentialEnergy(1E-4, units='Ha'),
+                 coords:  Optional[OptCoordinates] = None,
+                 n_cores: Optional[int] = None,
+                 **kwargs
+                 ) -> None:
+        """
+        Convenience function for constructing and running an optimiser
+
+        ----------------------------------------------------------------------
+        Arguments:
+            species (autode.species.Species):
+
+            method (autode.methods.Method):
+
+        Keyword Arguments
+            maxiter (int): Maximum number of iteration to perform
+
+            gtol (float | autode.values.GradientNorm): Tolerance on RMS(|∇E|)
+                 i.e. the root mean square of the gradient components. If
+                 a float then assume units of Ha Å^-1
+
+            etol (float | autode.values.PotentialEnergy): Tolerance on |∆E|
+                 between two consecutive iterations of the optimiser
+
+            kwargs (Any): Additional keyword arguments to pass to the
+                          constructor
+
+        Raises:
+            (ValueError | RuntimeError):
+        """
+
+        optimiser = cls(maxiter=maxiter, gtol=gtol, etol=etol, coords=coords,
+                        **kwargs)
+        optimiser.run(species, method, n_cores=n_cores)
+
+        return None
+
+    @property
+    def converged(self) -> bool:
+        """
+        Is this optimisation converged? Must be converged based on both energy
+        and gradient tolerance.
+
+        -----------------------------------------------------------------------
+        Returns:
+            (bool): Converged?
+        """
+        if self._abs_delta_e < self.etol / 10:
+            logger.warning(f'Energy change is overachieved. '
```

Comment:
Is this gonna show every time it converges?

Reply:
Should only hit convergence once, then break